main.js

```diff
@@ -0,0 +1,181 @@
+/*!
+GPII Electron
+Copyright 2016 Steven Githens
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+"use strict";
+
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+const {app, Menu, Tray} = require('electron')
+var path = require("path");
+var request = require("request");
+var os = require("os");
+
+// The tray needs to be global
+var tray = null;
+
+// Component to start and stop parts of the GPII as well as key in and key out users
+fluid.defaults("gpii.app", {
+    gradeNames: "fluid.modelComponent",
+    model: {
+        gpiiStarted: false,
+        keyedInSet: null
+    },
+    listeners: {
+        onCreate: [{
+            funcName: "gpii.app.startLocalFlowManager"
+        }, {
+            funcName: "{that}.changeStarted",
+            args: [true]
+        }]
+    },
+    modelListeners: {
+        keyedInSet: [{ //  Will this be a race condition?
+            funcName: "gpii.app.keyOut",
+            args: ["{change}.oldValue"]
+        }, {
+            funcName: "gpii.app.keyIn",
+            args: ["{change}.value"]
+        }]
+    },
+    invokers: {
+        changeStarted: {
+            changePath: "gpiiStarted",
+            value: "{arguments}.0"
+        }
+    }
+});
+
+gpii.app.startLocalFlowManager = function () {
+    var fluid = require("universal"),
+        gpii = fluid.registerNamespace("gpii");
+
+    if (os.platform() === "win32") {
+        var windows = require("gpii-windows/index.js");
+    }
+
+    gpii.start();
+};
+
+gpii.app.keyIn = function (token) {
+    request("http://localhost:8081/user/"+token+"/login", function(error, response, body) {
+        //TODO Put in some error logging
+    });
+}
+
+gpii.app.keyOut = function (token) {
+    request("http://localhost:8081/user/"+token+"/logout", function(error, response, body) {
+        //TODO Put in some error logging
+    });
+}
+
+
+// Component to create and update the task tray menu.
+fluid.defaults("gpii.taskTray", {
+    gradeNames: "fluid.modelComponent",
+    model: { // Do I need to specify a default or will it sync up with the other component?
```

Comment:
Yes! That was what I wanted. :) 
Fixed at ac8300c20e9da19d84f7d8b0368e521e8bea0a42